rs/registry/canister/src/registry.rs

```diff
@@ -284,30 +328,57 @@ impl Registry {
         self.apply_mutations_as_version(mutations, self.version);
     }
 
-    /// Verifies the implicit precondition corresponding to the mutation_type
-    /// field.
+    /// Some mutation_type(s) require that the key/value is currently (i.e.
+    /// right before the mutation is performed) present, while other(s) require
+    /// that the key/value is currently absent. This enforces those requirements.
     fn verify_mutation_type(&self, mutations: &[RegistryMutation]) -> Vec<Error> {
         mutations
             .iter()
-            .map(|m| {
-                let key = &m.key;
-                let latest = self
+            .map(|new_mutation| {
+                let RegistryMutation {
+                    key,
+                    mutation_type,
+                    value: _,
+                } = new_mutation;
+
+                let mutation_type = Type::try_from(*mutation_type).map_err(|err| {
+                    Error::MalformedMessage(format!("Unable to convert mutation_type: {}", err,))
+                })?;
+                let Some(requires_already_present) = mutation_type.requires_already_present()
+                else {
+                    return Ok(());
+                };
+
+                let already_present = self
```

Comment:
```suggestion
                let key_already_present = self
```

Reply:
well, `key` is maybe too specific, because key is present iff (corresponding) value is (also) present. If the goal here is for there to be a noun in the name, I'd go with `record` (which is what I've done).